src/lazy_buffer.rs

```diff
@@ -44,6 +40,18 @@ where
             }
         }
     }
+
+    pub fn prefill(&mut self, len: usize) {
+        if !self.done && len > self.buffer.len() {
+            for _ in 0..(len - self.buffer.len()) {
```

Comment:
could we just check for `!self.done` in line 45 and then iterate `for _ in len .. self.buffer.len()` in the next line, making the second check (`len > self.buffer.len()`) unnecessary?

Reply:
You're right that the second check can be removed.

It would need to be for _ in self.buffer.len()..len though (range expressions reversed). Would this be as readable in your opinion?

Reply:
> It would need to be for _ in self.buffer.len()..len though (range expressions reversed). Would this be as readable in your opinion?

I think `self.buffer.len()..len` is probably fine.